### PR TITLE
[SPARK-25456][SQL][TEST] Fix PythonForeachWriterSuite

### PR DESCRIPTION
**Pull request by squito**
PythonForeachWriterSuite was failing because RowQueue now needs to have a handle on a SparkEnv with a SerializerManager, so added a mock env with a serializer manager.

Also fixed a typo in the `finally` that was hiding the real exception.

Tested PythonForeachWriterSuite locally, full tests via jenkins.

### CODE REVIEW

File: sql/core/src/test/scala/org/apache/spark/sql/execution/python/PythonForeachWriterSuite.scala [hunk 1]
```diff
@@ -75,15 +78,20 @@ class PythonForeachWriterSuite extends SparkFunSuite with Eventually {
         tester = new BufferTester(memBytes, sleepPerRowReadMs)
         f(tester)
       } finally {
-        if (tester == null) tester.close()
+        if (tester != null) tester.close()
```
Comment (xuanyuanking) [inline sql/core/src/test/scala/org/apache/spark/sql/execution/python/PythonForeachWriterSuite.scala]: e... So here's why all test failed with NPE.

Comment (squito) [reply sql/core/src/test/scala/org/apache/spark/sql/execution/python/PythonForeachWriterSuite.scala]: yes, but that NPE just covered up another NPE from `SparkEnv.get.serializerManager`

**Comment by squito** [comment]
Jenkins, retest this please

**Comment by vanzin** [comment]
LGTM

**Comment by squito** [comment]
merged to master & 2.4

**Pull request closed without merging**